frontend/src/pages/NewAppointment.js

```diff
@@ -0,0 +1,142 @@
+import {Button, MenuItem, TextField} from '@mui/material';
+import {useEffect, useState} from 'react';
+import styled from 'styled-components/macro';
+import AdapterDateFns from '@mui/lab/AdapterDateFns';
+import {DatePicker, LocalizationProvider} from '@mui/lab';
+import moment from 'moment';
+import {getDoctors, submitAppointment} from '../service/DoctorApiService';
+
+export default function NewAppointment() {
+	const initialAppointment = {
+		date: null,
+		reasonForVisit: '',
+		notes: '',
+	};
+
+	const initialDoctor = {
+		id: '',
+		firstName: '',
+		lastName: '',
+		specialty: '',
+		street: '',
+		streetNumber: '',
+		postalCode: '',
+		city: '',
+		country: '',
+		phoneNumber: '',
+		mobileNumber: '',
+		emailAddress: '',
+		appointmentDto: initialAppointment,
+	};
+
+	const [newAppointment, setNewAppointment] = useState(initialAppointment);
+	const [doctor, setDoctor] = useState(initialDoctor);
+	const [doctors, setDoctors] = useState([]);
+
+	useEffect(() => {
+		getDoctors()
+			.then((doctors) => setDoctors(doctors))
+			.catch((error) => console.error(error.message));
+	}, []);
+
+	useEffect(() => {
+		setDoctor({...doctor, appointmentDto: newAppointment});
+	}, [newAppointment]);
+
+	const handleChange = (event) => {
+		setNewAppointment({
+			...newAppointment,
+			[event.target.name]: event.target.value,
+		});
+	};
+
+	const handleDateChange = (inputDate) => {
+		const formattedDate = moment(inputDate, 'YYYY-MM-DD').format('LL');
+		setNewAppointment({...newAppointment, date: formattedDate});
+	};
+
+	const handleDoctorChange = (event) => {
+		let doctorToUpdate = doctors.find(
```

Comment:
good to know, didn't know this existed. I always went the workaround of doing a filter and selecting element 0 of the array. have to adapt this in my code